### ISSUE TITLE
`torchaudio.compliance.kaldi.mfcc` seems to produce strange energy values.

### ISSUE THREAD

**Issue by cantabile-kwok** [pytorch/audio#2665]
### 🐛 Describe the bug

I am using torchaudio to extract Kaldi-like MFCC features for audio. I found `torchaudio.compliance.kaldi.mfcc` function which is exactly what I need. The code runs smoothly, but when I inspect the values, I found the energy produced by this function seems incorrect and very different to the results by Kaldi `compute-mfcc-feats` command.

Here is an example audio (LJ001-0110.wav): https://drive.google.com/file/d/1p6go649Li7tyCcwJN-d0Xy0uUvvfNCg2/view?usp=sharing

```python
import torchaudio
import matplotlib.pyplot as plt
audio, sr = torchaudio.load("LJ001-0110.wav")

mfcc = torchaudio.compliance.kaldi.mfcc(audio,
                                          sample_frequency=sr,
                                          frame_shift=256 * 1000 / sr,
                                          frame_length=1024 * 1000 / sr,
                                          use_energy=True,
                                          raw_energy=False,
                                          window_type='hanning')  # generates 13-dim mfcc with energy as 0-th dim
mfcc = mfcc.cpu().numpy()

# ====== plot values ========
plt.subplots(13, 1, figsize=(20, 65))
for i in range(13):
    plt.subplot(13, 1, i+1)
    plt.plot(mfcc[:, i])
    plt.title(f"{i}-th Dimension")
# plt.savefig(target_png, dpi=100)
plt.show()
```

And this generates this figure:
![torchaudio_mfcc](https://user-images.githubusercontent.com/58417810/189044649-8b4c0f91-b6c3-45cb-b541-44fa515a6b2b.png)

As we can see, the first dimension, which is the log-energy, seems to be 0 most of the time, which some steep peaks. I don't think this are the expected values, as this natural speech should have a complex energy trajectory. I also plot the Kaldi MFCC output as follows, under the same configuration:
![kaldi_mfcc](https://user-images.githubusercontent.com/58417810/189044667-455ad3cd-f5e3-4680-a79b-330610509ac2.png)

Clearly this looks more reasonable than torchaudio output. Also, the other dimensions of torchaudio output (except the 0-th) fit well to the Kaldi output. It is only the 0-th dimension that looks strange.

### Versions

Collecting environment information...
PyTorch version: 1.11.0+cu102
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: CentOS Linux 7 (Core) (x86_64)
GCC version: (GCC) 8.4.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.17

Python version: 3.9.6 (default, Aug 18 2021, 19:38:01)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-3.10.0-693.el7.x86_64-x86_64-with-glibc2.17
Is CUDA available: True
CUDA runtime version: 10.1.105
GPU models and configuration: 
GPU 0: GeForce RTX 2080 Ti
GPU 1: GeForce RTX 2080 Ti

Nvidia driver version: 460.73.01
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.21.2
[pip3] torch==1.11.0
[pip3] torchaudio==0.11.0
[pip3] torchtyping==0.1.4
[pip3] torchvision==0.12.0
[conda] numpy                     1.21.2                   pypi_0    pypi
[conda] torch                     1.11.0                   pypi_0    pypi
[conda] torchaudio                0.11.0                   pypi_0    pypi
[conda] torchtyping               0.1.4                    pypi_0    pypi
[conda] torchvision               0.12.0                   pypi_0    pypi

[conda] torchvision               0.12.0                   pypi_0    pypi

**Comment by hwangjeff** [comment]
Thanks for flagging this @cantabile-kwok. Could you also post the code you used to generate the Kaldi results?

**Comment by cantabile-kwok** [comment]
@hwangjeff  I used Kaldi command
```shell
compute-mfcc-feats --verbose=2 --config=conf/mfcc.conf scp,p:data/train/wav.scp \ 
                          ark,scp:exp/kaldi_mfcc/mfcc.ark,exp/kaldi_mfcc/mfcc.scp
```
where the mfcc.conf is a text file as follows:
```
--sample-frequency=22050
--frame-shift=11.6099774  
--frame-length=46.4399093
--use-energy=true
--raw-energy=false
--window-type=hanning
```
Note that 256 points in 22050 Hz approximates 11.6099774 miliseconds. Actually this doesn't affect the trajectory.

You might change `data/train/wav.scp` to any scp file pointing to this wav. Then I use [kaldiio](https://github.com/nttcslab-sp/kaldiio) to open the mfcc.ark and plot the figure with the same code as before.

**Comment by hwangjeff** [comment]
Running
```
import torchaudio
torchaudio.info("LJ001-0110.wav")
```
gives us
```
AudioMetaData(sample_rate=22050, num_frames=129437, num_channels=1, bits_per_sample=16, encoding=PCM_S)
```
i.e. the audio is formatted as 16-bit signed PCM.

By default, [`torchaudio.load`](https://pytorch.org/audio/main/backend.html#load) uses a value of `True` for parameter `normalize`. As a consequence, for a 16-bit PCM input, it actually normalizes the audio data by the maximum short value and returns the result as `float32`.

It doesn't look like Kaldi's `compute-mfcc-feats` performs this normalization. To achieve parity with Kaldi, then, we can disable normalization by providing `normalize=False` to `torchaudio.load`. Running the following
```
import torchaudio
import torch
import matplotlib.pyplot as plt

audio, sr = torchaudio.load("LJ001-0110.wav", normalize=False)
audio = audio.to(torch.float32)
mfcc = torchaudio.compliance.kaldi.mfcc(
    audio,
    sample_frequency=sr,
    frame_shift=256 * 1000 / sr,
    frame_length=1024 * 1000 / sr,
    use_energy=True,
    raw_energy=False,
    window_type="hanning",
)

# ====== plot values ========
plt.subplots(2, 1, figsize=(20, 65))
for i in range(2):
    plt.subplot(2, 1, i+1)
    plt.plot(mfcc[:, i])
    plt.title(f"{i}-th Dimension")
plt.show()
```
then produces the result you expect:
![Figure_1](https://user-images.githubusercontent.com/8345689/190287647-e19192dc-6fd4-41d4-a3a9-47bfbdc07ca5.png)

We can do a better job of flagging this behavior in our documentation. I'll take note.

Let us know if you have any other questions.

**Comment by cantabile-kwok** [comment]
@hwangjeff  I did not notice this difference, thanks a lot! It solved my confusion now :)

**Comment by hwangjeff** [comment]
@cantabile-kwok no problem — glad to hear it!
